packages/actor-init-sparql/lib/ActorInitSparql.ts

```diff
@@ -69,6 +71,28 @@ export class ActorInitSparql extends ActorInitSparqlBrowser {
     return existsSync(`${__dirname}/../test`);
   }
 
+  // Function to convert a URL like 'hypermedia@http://user:passwd@example.com to an IDataSource
+  public static getSourceObjectFromString(sourceString: string): {[id: string]: any} {
+    const source: {[id: string]: any} = {};
+    const mediaTypeRegex = /^[^:]*@/u;
+    const mediaTypeMatches = mediaTypeRegex.exec(sourceString);
+    if (mediaTypeMatches) {
+      source.type = mediaTypeMatches[0].slice(0, -1);
+      sourceString = sourceString.slice(sourceString.split('@')[0].length + 1);
```

Comment:
Split is not needed anymore. It is more performance to `slice` starting from the length of `source.type` (+1).